test/core/data.wast

```diff
@@ -158,44 +170,42 @@
 
 ;; Invalid bounds for data
 
-(assert_unlinkable
+(assert_trap
```

Comment:
Not sure why this is changed (maybe related to some new rules about out of bounds during instantiation https://github.com/WebAssembly/spec/blob/f607ec9a776a9e3425639f3cbe9b2d96bdc16b60/test/core/linking.wast#L395)

For us it's optional, but if we accept it, requires the change in spectest runner (see https://github.com/wasmx/fizzy/pull/750)